### PR TITLE
[#26] [#30] [Integrate] As a logged-in user, I can see the Survey Detail screen and submit answers

### LINKED ISSUE #26: [Integrate] As a logged-in user, I can see the Survey Detail screen
## Why
After selecting a survey from the Home screen, the user can see the Survey Detail screen and start the survey.

## Acceptance Criteria
- Show different types of survey questions together with the answers.
- The user can select their answer for each question.
- Unit tests are added and passed.

### LINKED ISSUE #30: [Integrate] As a logged-in user, I can submit my answers to the survey questions
## Why
The user can submit their answers to the survey questions.

## Acceptance Criteria
- The API endpoint `/api/v1/responses` is called.
  - If the API returns success, show the Complete Survey screen. Then after 3 seconds, the app will navigate to the Home screen.
  - If the API returns an error, show an error popup.
- Unit tests are added and passed.

## Design

Image

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
- Closes #26 
- Closes #30 

## What happened 👀

After selecting a survey from the Home screen, the user can see the Survey Detail screen and start the survey. After that the user can submit their answers to the survey questions.

## Insight 📝

- Show different types of survey questions together with the answers.
- The user can select their answer for each question.
- Unit tests are added and passed.

## Proof Of Work 📹

https://github.com/nimblehq/flutter-ic-kaung-thieu/assets/32578035/e33caa7b-348b-451c-a575-a9e0565a9833

### CODE REVIEW

File: lib/screens/detail/survey_detail_view_model.dart [hunk 1]
```diff
@@ -20,16 +27,40 @@ class SurveyDetailViewModel extends AutoDisposeAsyncNotifier<void> {
...
-  void submitSurvey() {
-    _isSubmitSuccess.add(true);
+  Future<void> submitSurvey(String surveyId) async {
+    final questionsRequest = _cache?.questions.map((question) =>
+        SurveyQuestionRequest(
+            id: question.id, answers: _findAnswers(question.answers)));
+    SubmitSurveyRequest request = SubmitSurveyRequest(
+        surveyId: surveyId, questions: questionsRequest?.toList() ?? []);
+    final result = await ref.read(submitSurveyUseCaseProvider).call(request);
+    if (result is Success) {
```
Comment (ducbm051291) [inline lib/screens/detail/survey_detail_view_model.dart:47]:
```suggestion
     _isSubmitSuccess.add(result is Success);
```

File: lib/screens/detail/survey_detail_view_model.dart [hunk 2]
```diff
@@ -77,161 +108,28 @@ class SurveyDetailViewModel extends AutoDisposeAsyncNotifier<void> {
...
-    );
-    Future.delayed(const Duration(seconds: 2), () {
-      _cache = detail;
-      _surveyDetail.add(detail);
-    });
+  Future<void> getSurveyDetail(String surveyId) async {
+    final result =
+        await ref.read(getSurveyDetailUseCaseProvider).call(surveyId);
+
+    if (result is Success) {
```
Comment (ducbm051291) [inline lib/screens/detail/survey_detail_view_model.dart:115]:
```suggestion
   if (result is Success<SurveyDetailDataResponse>) {
      final value = result.value.surveyDetailResponse?.toSurveyDetailModel();
```

**Review by ducbm051291 — APPROVED** [review]
The rest LGTM

File: lib/screens/detail/survey_detail_view_model.dart (hunk 2 above)
Comment (kaungkhantsoe) [reply lib/screens/detail/survey_detail_view_model.dart]: Fixed in https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/98/commits/0a6053c32062de94f76eb44ee708670729092373

**Comment by manh-t** [comment]
@kaungkhantsoe are you aware of this UI error? I face it on my device when navigating to the Question screen
```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following assertion was thrown while applying parent data.:
Incorrect use of ParentDataWidget.
The ParentDataWidget Expanded(flex: 1) wants to apply ParentData of type FlexParentData to a
RenderObject, which has been set up to accept ParentData of incompatible type BoxParentData.
Usually, this means that the Expanded widget has the wrong ancestor RenderObjectWidget. Typically,
Expanded widgets are placed directly inside Flex widgets.
The offending Expanded is currently placed inside a Center widget.
The ownership chain for the RenderObject that received the incompatible parent data was:
  SizedBox.shrink ← Expanded ← Consumer ← Center ← Expanded ← Column ← SurveyQuestionContent ←
RepaintBoundary ← IndexedSemantics ← _SelectionKeepAlive ← ⋯

When the exception was thrown, this was the stack:
#0      RenderObjectElement._updateParentData.<anonymous closure> (package:flutter/src/widgets/framework.dart:6042:11)
#1      RenderObjectElement._updateParentData (package:flutter/src/widgets/framework.dart:6059:6)
#2      RenderObjectElement.attachRenderObject (package:flutter/src/widgets/framework.dart:6082:7)
#3      RenderObjectElement.mount (package:flutter/src/widgets/framework.dart:5751:5)
#4      SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6299:11)
...     Normal element mounting (24 frames)
#28     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3863:16)
#29     MultiChildRenderObjectElement.inflateWidget (package:flutter/src/widgets/framework.dart:6435:36)
#30     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6447:32)
...     Normal element mounting (49 frames)
#79     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3863:16)
#80     Element.updateChild (package:flutter/src/widgets/framework.dart:3592:18)
#81     SliverMultiBoxAdaptorElement.updateChild (package:flutter/src/widgets/sliver.dart:1405:37)
#82     SliverMultiBoxAdaptorElement.createChild.<anonymous closure> (package:flutter/src/widgets/sliver.dart:1390:20)
#83     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2605:19)
#84     SliverMultiBoxAdaptorElement.createChild (package:flutter/src/widgets/sliver.dart:1382:12)
#85     RenderSliverMultiBoxAdaptor._createOrObtainChild.<anonymous closure> (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:353:23)
#86     RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:2246:59)
#87     PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:1035:15)
#88     RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:2246:14)
#89     RenderSliverMultiBoxAdaptor._createOrObtainChild (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:342:5)
#90     RenderSliverMultiBoxAdaptor.insertAndLayoutChild (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:490:5)
#91     RenderSliverFixedExtentBoxAdaptor.performLayout (package:flutter/src/rendering/sliver_fixed_extent_list.dart:250:17)
#92     RenderObject.layout (package:flutter/src/rendering/object.dart:2135:7)
#93     RenderSliverEdgeInsetsPadding.performLayout (package:flutter/src/rendering/sliver_padding.dart:149:12)
#94     _RenderSliverFractionalPadding.performLayout (package:flutter/src/widgets/sliver_fill.dart:166:11)
#95     RenderObject.layout (package:flutter/src/rendering/object.dart:2135:7)
#96     RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:516:13)
#97     RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1600:12)
#98     RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1507:20)
#99     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1973:7)
#100    PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:999:18)
#101    RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:513:19)
#102    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:884:13)
#103    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:378:5)
#104    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1175:15)
#105    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1104:9)
#106    SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1015:5)
#107    _invoke (dart:ui/hooks.dart:148:13)
#108    PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:318:5)
#109    _drawFrame (dart:ui/hooks.dart:115:31)
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by manh-t** [comment]
also, there is an overflow error if the text is long
image

**Comment by kaungkhantsoe** [comment]
> also, there is an overflow error if the text is long

@manh-t I will fix this error in a separate bug ticket 🙏

**Comment by kaungkhantsoe** [comment]
> are you aware of this UI error? I face it on my device when navigating to the Question screen

@manh-t Not sure how you got this error because I am seeing everything went well on my device. Did you face a crash or UI error while you see that error on your device?

**Comment by manh-t** [comment]
> @manh-t Not sure how you got this error because I am seeing everything went well on my device. Did you face a crash or UI error while you see that error on your device?

The UI displays well but there is an implicit error related to widget tree in the log. You might have to check the log as well. FYI: my device is emulator Pixel 5 API 29

**Comment by kaungkhantsoe** [comment]
> > @manh-t Not sure how you got this error because I am seeing everything went well on my device. Did you face a crash or UI error while you see that error on your device?
> 
> The UI displays well but there is an implicit error related to widget tree in the log. You might have to check the log as well. FYI: my device is emulator Pixel 5 API 29

Fixed the error in https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/102/commits/a0b1831183765e1f4c906e4b19d6797e21cdaa81

**Review by manh-t — APPROVED** [review]
All errors have been fixed in other PRs. Hence, approved 🚀

**Pull request merged**